Inconsistent behaviour of `Indentation` check

I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Example.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

/var/tmp $ cat Example.java
import java.util.List;

class Example {
    static void main() {
        List.of("foo").stream()
            .findFirst()
            .ifPresentOrElse(
                System.out::println,
                () -> {
                    throw new IllegalStateException();
                }
            );
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-12.1.2-all.jar -c config.xml Example.java
Starting audit...
[ERROR] /var/tmp/Example.java:12:13: 'method call rparen' has incorrect indentation level 12, expected level should be 8. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

There should be no errors.

In `AnotherExample.java` I've replaced `List.of("foo").stream()` with `Stream.of("foo")` and there are no errors:

```bash
/var/tmp $ javac AnotherExample.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

/var/tmp $ cat AnotherExample.java
import java.util.stream.Stream;

class AnotherExample {
    static void main() {
        Stream.of("foo")
            .findFirst()
            .ifPresentOrElse(
                System.out::println,
                () -> {
                    throw new IllegalStateException();
                }
            );
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-12.1.2-all.jar -c config.xml AnotherExample.java
Starting audit...
Audit done.
```

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java
@@ -217,9 +217,38 @@ public class MethodCallHandler extends AbstractExpressionHandler {
                     new IndentLevel(getIndent(), getBasicOffset()),
                     false, true);
 
-                checkRightParen(lparen, rparen);
+                checkRparenIndent(lparen, rparen);
                 checkWrappingIndentation(getMainAst(), getCallLastNode(getMainAst()));
             }
+        }
+    }
+
+    /**
+     * Checks the indentation of the right parenthesis for method calls.
+     *
+     * @param lparen left parenthesis associated with rparen
+     * @param rparen parenthesis to check
+     */
+    private void checkRparenIndent(DetailAST lparen, DetailAST rparen) {
+        final int rparenLevel = expandedTabsColumnNo(rparen);
+        final int lparenLevel = expandedTabsColumnNo(lparen);
+
+        final IndentLevel standardIndent = getIndent();
+
+        // For chained method calls, also allow rparen at the line start position
+        IndentLevel enhancedIndent = standardIndent;
+        if (getParent() instanceof MethodCallHandler) {
+            final int lineStart = getLineStart(getFirstAst(getMainAst()));
+            if (lineStart != standardIndent.getFirstIndentLevel()) {
+                enhancedIndent = IndentLevel.addAcceptable(standardIndent,
+                        new IndentLevel(lineStart));
+            }
+        }
+
+        if (rparenLevel != lparenLevel + 1
+                && !enhancedIndent.isAcceptable(rparenLevel)
+                && isOnStartOfLine(rparen)) {
+            logError(rparen, "rparen", rparenLevel);
         }
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -349,6 +349,7 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             "53:19: " + getCheckMessage(MSG_CHILD_ERROR, "method call", 18, 20),
             "54:15: " + getCheckMessage(MSG_ERROR, "method call rparen", 14, 16),
             "75:13: " + getCheckMessage(MSG_ERROR, "lambda arguments", 12, 16),
+            "81:16: " + getCheckMessage(MSG_ERROR, "method call rparen", 15, 8),
         };
         verifyWarns(checkConfig, getPath("InputIndentationMethodCallLineWrap.java"), expected);
     }
@@ -1601,10 +1602,10 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         checkConfig.addProperty("throwsIndent", "4");
         final String fileName = getPath("InputIndentationChainedMethodCalls.java");
         final String[] expected = {
-            "33:5: " + getCheckMessage(MSG_CHILD_ERROR, "method call", 4, 8),
-            "38:5: " + getCheckMessage(MSG_ERROR, ".", 4, 8),
-            "39:5: " + getCheckMessage(MSG_ERROR, ".", 4, 8),
-            "42:5: " + getCheckMessage(MSG_ERROR, "new", 4, 8),
+            "36:5: " + getCheckMessage(MSG_CHILD_ERROR, "method call", 4, 8),
+            "41:5: " + getCheckMessage(MSG_ERROR, ".", 4, 8),
+            "42:5: " + getCheckMessage(MSG_ERROR, ".", 4, 8),
+            "45:5: " + getCheckMessage(MSG_ERROR, "new", 4, 8),
         };
         verifyWarns(checkConfig, fileName, expected);
     }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationChainedMethodCalls.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationChainedMethodCalls.java
@@ -14,6 +14,9 @@
 
 package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;     //indent:0 exp:0
 
+import java.util.List;                                                      //indent:0 exp:0
+import java.util.stream.Stream;                                             //indent:0 exp:0
+
 class InputIndentationChainedMethodCalls {                                  //indent:0 exp:0
 
   public InputIndentationChainedMethodCalls(Object... params) {             //indent:2 exp:2
@@ -43,5 +46,37 @@ class InputIndentationChainedMethodCalls {                                  //in
         .getInstance()                                                      //indent:8 exp:8
         .doNothing("nothing")                                               //indent:8 exp:8
         .length();                                                          //indent:8 exp:8
+
+    new InputIndentationChainedMethodCalls("param1", "param2")              //indent:4 exp:4
+        .getInstance().doNothing(                                           //indent:8 exp:8
+            "nothing",                                                      //indent:12 exp:12
+            new InputIndentationChainedMethodCalls()                        //indent:12 exp:12
+                .getInstance()                                              //indent:16 exp:16
+                .doNothing("abc")                                           //indent:16 exp:16
+        );                                                                  //indent:8 exp:8
+  }                                                                         //indent:2 exp:2
+
+  public static void testListOfStreamPattern() {                            //indent:2 exp:2
+    List.of("foo").stream()                                                 //indent:4 exp:4
+        .findFirst()                                                        //indent:8 exp:8
+        .ifPresentOrElse(                                                   //indent:8 exp:8
+          System.out::println,                                              //indent:10 exp:10
+          () -> {                                                           //indent:10 exp:10
+            throw new IllegalStateException();                              //indent:12 exp:12
+          }                                                                 //indent:10 exp:10
+        );                                                                  //indent:8 exp:8
+
+    List.of("a", "b", "c").stream()                                         //indent:4 exp:4
+        .filter(s -> s.length() > 1)                                        //indent:8 exp:8
+        .forEach(System.out::println);                                      //indent:8 exp:8
+
+    Stream.of("foo")                                                        //indent:4 exp:4
+        .findFirst()                                                        //indent:8 exp:8
+        .ifPresentOrElse(                                                   //indent:8 exp:8
+          System.out::println,                                              //indent:10 exp:10
+          () -> {                                                           //indent:10 exp:10
+            throw new IllegalStateException();                              //indent:12 exp:12
+          }                                                                 //indent:10 exp:10
+        );                                                                  //indent:8 exp:8
   }                                                                         //indent:2 exp:2
 }                                                                           //indent:0 exp:0

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationMethodCallLineWrap.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationMethodCallLineWrap.java
@@ -74,4 +74,10 @@ public class InputIndentationMethodCallLineWrap {                            //i
             chainingWithLambda(                                              //indent:12 exp:12
             x -> x);                                                         //indent:12 exp:16 warn
     }                                                                        //indent:4 exp:4
+
+
+    void testLineWrap() {                                                    //indent:4 exp:4
+        toString(                                                            //indent:8 exp:8
+               );                                                            //indent:15 exp:8 warn
+    }                                                                        //indent:4 exp:4
 }                                                                            //indent:0 exp:0

